pkg/hs/urbit-king/lib/Urbit/Vere/Ames/UDP.hs

```diff
@@ -5,7 +5,8 @@
 
     - If can't open the port, wait and try again repeatedly.
     - If there is an error reading or writting from the open socket,
-      close it and open another.
+      close it and open another, making sure, however, to reuse the
+      same port
```

Comment:
What kinds of errors make it necessary to rebind the socket?

Reply:
@pilfer-pandex bump.

Reply:
Any failure to send along the socket, whatsoever.

Reply:
I see that's what it does, but i'm trying to understand why. I'm not aware of any issues with UDP itself that would require you to rebind the socket (unless we were binding a specific ip address and that address changed, but we always use localhost or "any address").

Reply:
@pilfer-pandex rebump.

Reply:
@joemfb I understand what you're saying here: The udp socket, we think, is a "stateless" object, such that destroying it and creating it anew is exactly the same thing as using the existing one. That makes sense as a hypothesis, but on the other hand, I can't believe that benjamin didn't know about it. The code here was very deliberately written to do this cumbersome thing, and the comment at the top of the file calls out how important it is to do this (but not why). I don't feel comfortable removing it until I understand what problem it solves. If you could conclusively argue that it definitely has no effect, then I'd be less concerned. Right now, any "IOError" that can occur on send will trigger this error handling. I think (?) that all these IOErrors correspond to potential errnos. You think (?) that all of those possible errnos are independent of any "state" of the socket. These (?)s need to be spelled out in full.